walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,60 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Collections;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  public static final Comparator<Event> orderByStartThenEnd = new Comparator<Event>() {
+    @Override
+    public int compare(Event a, Event b){
+      if(a.getWhen().start() != b.getWhen().start())
+        return Integer.compare(a.getWhen().start(), b.getWhen().start());
+      else
+        return Integer.compare(a.getWhen().end(), b.getWhen().end());
+    }
+  };
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> peopleAttending = request.getAttendees();
+    List<Event> eventWithPeopleAttending = new ArrayList<>();
+    for(Event e : events){
+      boolean is_not_attending = true;
+      for(String x : e.getAttendees()){
+        if(peopleAttending.contains(x))
+          is_not_attending = false;
```

Comment:
nit: should we follow a single naming pattern likeThis or like_this?

Reply:
Sorry I forget everytime :(. I will make sure the next time.